Automattic/wp-calypso
# Pattern Library: Use custom function for setting canonical URL

Fixes https://github.com/Automattic/dotcom-forge/issues/6441

## Proposed Changes

We already have `<link rel=canonical>` tags in the pattern library (introduced in #87726), but the URLs we output don't strip out query parameters. This PR changes that. Query parameters are now always removed from the canonical tags.

Why do we want this? The content on `/patterns/about` and `/patterns/about?grid=1` is exactly the same, and canonical tags are meant to prevent issues with duplicate content. Besides the `grid` query param, the same is also true for the search `s` param. All patterns that can be accessed through search can also be accessed through the regular categories.

## Testing Instructions



1. Open an incognito window
2. Navigate to `/patterns/about?grid=1`
3. View the page source
4. Ensure that the canonical tag is rendered like this: `<link rel="canonical" href="https://wordpress.com/patterns/about"/>`
